### PR TITLE
Add new gates to Photonics simulator

### PR DESCRIPTION
**Pull request by Omar-ORCA**
* Added create and annihilate photonic gates
* Updated examples

### CODE REVIEW

File: docs/sphinx/api/default_ops.rst [hunk 1]
```diff
@@ -638,31 +638,67 @@ operations, each operating on 2 qubits.
 Photonic Operations on Qudits
 =============================
 
-These operations are valid only on the `photonics` target which does not support the quantum operations above.
+These operations are valid only on the `photonics` target which does not support
+the quantum operations above.
 
-:code:`plus`
+:code:`create`
```
Comment (khalatepradnya) [inline docs/sphinx/api/default_ops.rst:644]: @bettinaheim / @1tnguyen : Thoughts on this gate naming? I think it is also a built-in `Operator` for upcoming feature.
I think it _means_ the same would like to confirm that there isn't an issue...

File: docs/sphinx/api/default_ops.rst [hunk 2]
```diff
@@ -638,31 +638,67 @@ operations, each operating on 2 qubits.
...
 
 .. tab:: C++
 
     .. code-block:: cpp
 
         cudaq::qvector<3> q(1);
-        plus(q[0]);
+        create(q[0]);
+
+:code:`annihilate`
```
Comment (khalatepradnya) [inline docs/sphinx/api/default_ops.rst:669]: Ditto

File: docs/sphinx/api/default_ops.rst (hunk 1 above)
Comment (khalatepradnya) [inline docs/sphinx/api/default_ops.rst:644]: These names don't match the RFC here https://github.com/NVIDIA/cuda-quantum/issues/1937. 
@Omar-ORCA : Can you please add the proposal to RFC?

Comment (Omar-ORCA) [reply docs/sphinx/api/default_ops.rst:644]: I added the proposal as a comment

**Review by khalatepradnya — APPROVED** [review]
LGTM.

**Review by bettinaheim — APPROVED** [review]
I only looked at the files under code ownership - Pradnya reviewed the rest.

**Pull request merged**